vanetza/security/ecc_point_decompression_visitor.hpp

```diff
@@ -0,0 +1,34 @@
+#include <boost/variant.hpp>
+#include <vanetza/security/ecc_point.hpp>
+
+namespace vanetza
+{
+namespace security
+{
+
+struct EccPointDecompressionVisitor : public boost::static_visitor<Uncompressed> {
+    Uncompressed operator()(X_Coordinate_Only p) {
+        // Without the additional bit of the y coordinate, we cannot restore y.
```

Comment:
Returning incomplete data seems wrong to me.

Reply:
The memo you referenced does not use all possible keys/EC points but only those whose y coordinate lies within the lower half of the field, i.e. `y <= (p-1) / 2` resp. `y == min(y, p-y)`. The memo's decompression does not require an additional bit of information in the _compressed form_ of the EC point because the memo defines that information in advance by using only half of the possible EC points.

Since the common representation of (possibly compressed) EC points includes the compression type in the first byte anyway, I don't see a point in omitting the LSB_Y information by using compression type 0 (X_Coordinate_Only).

Concerning our decompression implementation: We cannot make the assumption that all keys we encounter have this property and therefore we cannot decompress *all* keys only from their x coordinate. So either we return incomplete data, throw an error and exit, or throw an error silently (are there any possibilities for that in vanetza?) and the security entity cannot use that key further.

Reply:
I would rather not call exit or ignore an error state silently. Either throw an exception (if it is really exceptional) or use boost::optional or alike to represent a failure (if it is common that an action fails).